src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py

```diff
@@ -0,0 +1,313 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+# noqa: E501
+import asyncio
+import logging
+from typing import Any, Callable, Dict, List
+
+from tqdm import tqdm
+
+from promptflow._sdk._telemetry import ActivityType, monitor_operation
+
+from ._conversation import CallbackConversationBot, ConversationBot, ConversationRole
+from ._conversation._conversation import simulate_conversation
+from ._model_tools import (
+    CONTENT_HARM_TEMPLATES_COLLECTION_KEY,
+    AdversarialTemplateHandler,
+    AsyncHTTPClientWithRetry,
+    ManagedIdentityAPITokenManager,
+    ProxyChatCompletionsModel,
+    RAIClient,
+    TokenScope,
+)
+from ._utils import JsonLineList
+
+logger = logging.getLogger(__name__)
+
+
+class AdversarialSimulator:
+    @monitor_operation(activity_name="adversarial.simulator.init", activity_type=ActivityType.PUBLICAPI)
+    def __init__(self, *, template: str, project_scope: Dict[str, Any]):
+        """
+        Initializes the adversarial simulator with a template and project scope.
+
+        :param template: Template string used for generating adversarial inputs.
+        :type template: str
+        :param project_scope: Dictionary defining the scope of the project. It must include the following keys:
+            - "subscription_id": Azure subscription ID.
+            - "resource_group_name": Name of the Azure resource group.
+            - "workspace_name": Name of the Azure Machine Learning workspace.
+            - "credential": Azure credentials object for authentication.
+        :type project_scope: Dict[str, Any]
+        """
+        if template not in CONTENT_HARM_TEMPLATES_COLLECTION_KEY:

```

Comment:
This hopefully will be an API call right? So we don't need to release new SDK versions for new templates added on the service side.

Reply:
Removed this strict check.